### PR TITLE
Mak265 ts dev

### PR DESCRIPTION
**Pull request by mkoma808**
Fixed the code coverage issues and added rudimentary interval data visualization.

### CODE REVIEW

**Comment by ets994** [comment]
87% coverage, hell yeah

File: src/App.tsx [hunk 1]
```diff
@@ -13,7 +11,7 @@ const App: React.FC = () => {
         fileType={FileType.csv}
         onChange={function() {}}
       />
-      <TimelineComponent data={new Data('path/to/file', [{}])} />
+      {/* <TimelineComponent data={new Data('path/to/file', [{}])} />*/}
```
Comment (eileenvanheerde) [inline src/App.tsx]: Why did you comment out the TimelineComponent?

Comment (mkoma808) [reply src/App.tsx]: The TimelineComponent is rendered in the ParseComponent, so it should not be in the App.tsx file.

File: src/__tests__/TimelineComponent.test.tsx [hunk 2]
```diff
@@ -60,19 +314,72 @@ describe('<TimelineComponent /> Unit Tests', () => {
...
+
+      wrapper.find('#x2Select')
+          .simulate('change', {target: {value: 'Order Date'}});
+
+      expect(updateBarsSpy).toHaveBeenCalled();
     });
   });
 
   describe('updateBars()', () => {
     it('dummy test', () => {
```
Comment (camilleion) [inline src/__tests__/TimelineComponent.test.tsx]: I think you forgot to change 'dummy test'

File: src/components/ParserComponent.tsx [hunk 3]
```diff
@@ -164,22 +227,34 @@ export default class ParserComponent extends React.Component<ParserInterface,
...
    * @param {Object} fileEvent: the event passed into this component
    */
-  parse(fileEvent: any) {
-    // this.isValid(fileEvent);
-    // this.sortData(this.state.data);
-    // this.inferTypes(this.state.data);
+  async parse(fileEvent: any) {
+    this.setState(() => {
+      return {
+        showTimeline: false,
```
Comment (camilleion) [inline src/components/ParserComponent.tsx]: show timeline instantiates to false and then is true if the file is a csv? What if it is not a csv?

File: src/components/TimelineComponent.tsx [hunk 4]
```diff
@@ -25,20 +59,187 @@ export default class TimelineComponent extends
...
   }
 
+
   /**
    * Purpose: waits until the component has properly mounted before drawing the
    * timeline
    */
   componentDidMount(): void {
+    // console.log(this.state.data);
+
```
Comment (camilleion) [inline src/components/TimelineComponent.tsx]: for longer functions such as componentDidMount() some more inline commenting would be usefule

File: src/components/TimelineComponent.tsx [hunk 5]
```diff
@@ -218,129 +475,332 @@ export default class TimelineComponent extends
...
+   */
+  updateChart() {
+    // recover the new scale
+    if (d3.event !== null) {
+      scale = d3.event.transform.k;
+    } else {
+      console.warn('d3.event was null');
+    }
+
+    // const newX = d3.event.transform.rescaleX(x);
```
Comment (camilleion) [inline src/components/TimelineComponent.tsx]: delete these?

File: src/components/TimelineComponent.tsx [hunk 6]
```diff
@@ -218,129 +475,332 @@ export default class TimelineComponent extends
...
+            scale * timeScale(new Date(d[xColumn])))
+          .attr('width', (d: any, i: number) =>
+            scale * (timeScale(new Date(d[xColumn2])) -
+              timeScale(new Date(d[xColumn]))));
+
+      d3.selectAll('.xtick')
+          .attr('transform', (d: any, i: number) =>
+            `translate(${scale * timeScale(new Date(d.text))},${height})`);
     }
+    // console.warn(d3.selectAll('.bar').size());
```
Comment (camilleion) [inline src/components/TimelineComponent.tsx]: delete this?

File: src/components/TimelineComponent.tsx [hunk 7]
```diff
@@ -218,129 +475,332 @@ export default class TimelineComponent extends
...
+        .on('mouseleave', this.ttLeave);
+  }
 
-    /**
-     * Purpose: called when the timeline is dragged by the user
-     */
-    function dragged() {
-      ttUpdatePos(d3.event.sourceEvent.x, d3.event.sourceEvent.y);
+  /**
+   *
```
Comment (camilleion) [inline src/components/TimelineComponent.tsx]: more extensive comments/docstring here

File: src/components/TimelineComponent.tsx [hunk 8]
```diff
@@ -218,129 +475,332 @@ export default class TimelineComponent extends
...
+        .on('mouseleave', this.ttLeave);
+  }
 
-      deltaX += d3.event.sourceEvent.movementX;
-      if (deltaX > 0) {
-        deltaX = 0;
+  /**
+   * Purpose: used to update which bars are being rendered to the screen
+   */
+  updateBars() {
```
Comment (camilleion) [inline src/components/TimelineComponent.tsx]: Long method: would be a good method for some in-line commenting

File: src/components/TimelineComponent.tsx [hunk 9]
```diff
@@ -218,129 +475,332 @@ export default class TimelineComponent extends
...
+                  .attr('dy', '.15em')
+                  .attr('transform', 'rotate(-90)');
+            },
+            (update: any) => update,
+            (exit: { remove: () => void; }) => exit.remove()
+        );
+  }
+
+  /**
+   *
```
Comment (camilleion) [inline src/components/TimelineComponent.tsx]: missing docstring?

**Pull request merged**